chainer/datasets/cifar.py

```diff
@@ -130,20 +134,21 @@ def load(archive, file_name):
 
     raw = download.cache_or_load_file(npz_path, creator, numpy.load)
     train = _preprocess_cifar(raw['train_x'], raw['train_y'], withlabel,
-                              ndim, scale)
+                              ndim, scale, dtype)
     test = _preprocess_cifar(raw['test_x'], raw['test_y'], withlabel, ndim,
-                             scale)
+                             scale, dtype)
     return train, test
 
 
-def _preprocess_cifar(images, labels, withlabel, ndim, scale):
+def _preprocess_cifar(images, labels, withlabel, ndim, scale, dtype):
     if ndim == 1:
         images = images.reshape(-1, 3072)
     elif ndim == 3:
         images = images.reshape(-1, 3, 32, 32)
     else:
         raise ValueError('invalid ndim for CIFAR dataset')
-    images = images.astype(numpy.float32)
+    dtype = chainer._get_dtype(dtype)
```

Comment:
Need `import chainer` and `_get_dtype` -> `get_dtype`

Reply:
fixed